lib/maze/client/bb_api_client.rb

```diff
@@ -106,6 +132,22 @@ def query_api(path, query=nil, uri=USER_SPECIFIC_URI)
 
         JSON.parse(res.body)
       end
+
+      # Queries the BitBar REST API with the DELETE method
+      def delete_query_api(path, query=nil, uri=USER_SPECIFIC_URI)
+        if query
+          encoded_query = URI.encode_www_form(query)
+          uri = URI("#{uri}/#{path}?#{encoded_query}")
+        else
+          uri = URI("#{uri}/#{path}")
+        end
+        request = Net::HTTP::Delete.new(uri)
+        request.basic_auth(@access_key, '')
+        res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+          http.request(request)
+        end
+        res
+      end
```

Comment:
I think this is identical to `query_api` apart from the method being used?  I'd suggest adding a new optional `method` parameter to that instead, defaulting to`:get`.  Then have and if/else or case when creating the `request`.